lib/ansible/modules/packaging/os/dnf.py

```diff
@@ -314,11 +333,28 @@ def _install_remote_rpms(base, filenames):
         base.package_install(pkg)
 
 
-def ensure(module, base, state, names):
+def ensure(module, base, state, names, autoremove):
     # Accumulate failures.  Package management modules install what they can
     # and fail with a message about what they can't.
     failures = []
     allow_erasing = False
+
+    if autoremove is not None:
+        if state not in ["absent", None]:
+            module.fail_json(msg="Autoremove should be used alone or with state=absent")
```

Comment:
We should do this check in main() as well.  There's not much being done before this is called (unlike the __version__ check I mentioned below) but it makes organizational sense to put all the validation of module parameters in one place.  That's both traditionally in main() after instantiating the AnsibleModule() and since we're going to do the version check there, we should do this there as well.